src/bentoml/_internal/cloud/bentocloud.py

```diff
@@ -663,38 +663,37 @@ def _do_push_model(
                     transmission_strategy = "presigned_url"
                     presigned_upload_url = remote_model.presigned_upload_url
 
-        io_mutex = threading.Lock()
-
         def io_cb(x: int):
-            with io_mutex:
-                self.spinner.transmission_progress.update(upload_task_id, advance=x)
+            self.spinner.transmission_progress.update(upload_task_id, advance=x)
 
-        with CallbackIOWrapper(read_cb=io_cb) as tar_io:
+        with NamedTemporaryFile(
```

Comment:
Should we use `SpooledTemporaryFile` instead?

Reply:
NamedTemporyFile isn't a in-memory approach when `dir` points to a custom location. But `SpooledTemporaryFile` keeps file in memory with a max size, I don't see much benefit because it must be saved to disk in the later multipart uploading step.

Reply:
Why does it have to be saved?

Reply:
Because we need to split it into several chunks to be read from multi-thread. The idea is we don't want to load the whole chunk into memory, so we save the file and read it with separate file pointers.

Reply:
Seems like we could use the buffer like we did before if it's still in memory, but maybe that's too much work for a temporary fix.

Reply:
That will be an implementation detail hidden behind `SpooledTemporaryFile` and we don't need to change other code.

Reply:
Yes, making such distinctions will make the code verbose and ugly.